docs/source/paper.md

```diff
@@ -0,0 +1,102 @@
+[Materials Cloud](https://www.materialscloud.org/).
+One of the many challenges for such frameworks is maximising fault-tolerance whilst simultaneously maintaining high-throughput, 
+often across several systems (typically the client launching the tasks, the supercomputer carrying out the computations
+and the server hosting the database).  
+
+On the software level these problems are perhaps best addressed by using messaging
+brokers that take responsibility for guaranteeing the durability and atomicity of messages and often enable event based
+communication.  Indeed solutions such as [RabbitMQ](https://www.rabbitmq.com/) see widespread adoption in industry, however adoption
+in academia has been more limited, with home-made queue data structure, race condition susceptible locks and polling based
+solutions common. This is likely due to the relatively complex nature of some of the APIs (which
+reflect the non-trivial nature of the underlying protocol), and a lack of familiarity with event-based programming in
+general within the community.  [KiwiPy](https://kiwipy.readthedocs.io/en/latest/) was designed specifically to address 
+both these potential issues and to provide an as simple as possible interface for building robust event based systems.
+
+KiwiPy provides three main message types to the user, task queues, Remote Procedure Calls (RPCs), and, broadcasts.
+These are all exposed via one class called the `Communicator` which can be trivially constructed by providing a URI
+string pointing to the RabbitMQ server, this is in contrast to the several steps needed to establish a connection to 
+perform each of the message types listed above using [standard libraries](https://www.rabbitmq.com/getstarted.html).
+Furthermore, by default, kiwiPy creates a separate communication thread that the user never sees allowing them to 
+interact with the communicator using familiar python syntax without the need to be familiar with either coroutines or
+multithreading.  This has the knock-on advantage that kiwiPy will maintain heartbeats (periodic check to make sure the
+connection is still alive) with the server whilst the user code can be doing other things.
+Heartbeats are an essential part of RabbitMQ's fault tolerance whereby two missed checks will automatically trigger the
+message to be requeued to be picked up by another client.
+
+To demonstrate some of the possible usage scenarios we briefly outline the way kiwiPy is used in AiiDA.  AiiDA, amongst 
+other things, manages the execution of complex workflows made up of processes which may have checkpoints.
+
+Task queues
+-----------
+
+As with any high-throughput workflow engine AiiDA maintains a queue to which processes are submitted (typically from the
+user's workstation).  Tasks are then consumed by multiple daemon processes (which may also be on the user's workstation
+or remote).  The daemon can be gracefully or abruptly shut down and no task will be lost.  Furthermore, there are no
```

Comment:
```suggestion
or remote) and will only be removed from the task queue once they have been acknowledged to be completed by the consumer.  The daemon can be gracefully or abruptly shut down and no task will be lost, since the task will simply be requeued by the broker once it notices that the consumer has died.  Furthermore, there are no
```